src/exercise/client/src/App.js

```diff
@@ -2,59 +2,32 @@ import "./App.css";
 import { DialogContentContainer } from "monday-ui-react-core";
 import "monday-ui-react-core/dist/main.css";
 import Header from "./Components/Header/Header";
-import InputItem from "./Components/InputItem/InputItem";
-import BottomButtons from "./Components/BottomButtons/BottomButtons";
-import { useCallback, useEffect, useState, useRef } from "react";
-import TaskList from "./Components/TaskList/TaskList";
-import {
-  getTasks,
-  deleteTask,
-  changeTaskStatus,
-  addTask,
-} from "./service/item_client";
+import InputItem from "./Components/InputItem/InputItem-connector";
+import BottomButtons from "./Components/BottomButtons/BottomButtons-connector";
+import TaskList from "./Components/TaskList/TaskList-connector";
+import { useState, useRef } from "react";
+import { ToastContainer, Zoom } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import SearchBar from "./Components/FilterBar/FilterBar";
 
 function App() {
-  const [tasks, setTasks] = useState([]);
-
-  async function updateTasks() {
-    const newTasks = await getTasks();
-    setTasks(newTasks.data);
-  }
-
-  useEffect(() => {
-    updateTasks();
-  }, []);
-
-  const handleDelete = useCallback(async (taskId) => {
-    setTasks(tasks.filter((task) => task.taskId !== taskId));
-    await deleteTask(taskId);
-  });
-
-  const handleDeleteAll = useCallback(async () => {
-    taskSections.current = [];
-    setTasks([]);
-    await deleteTask("all");
-  });
-
   const taskSections = useRef([]);
+  const [searchFilter, setSearchFilter] = useState("");
```

Comment:
the search was meant to be performed within the redux module - in the reducer
in this case you are not utilizing the benefits redux brings you